community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_0/TransactionBoundQueryContext.scala

```diff
@@ -191,7 +191,12 @@ class TransactionBoundQueryContext(graph: GraphDatabaseAPI, tx: Transaction, sta
       statement.dataWriteOperations().relationshipSetProperty(id, properties.Property.property(propertyKeyId, value) )
     }
 
-    def getById(id: Long) = graph.getRelationshipById(id)
+    def getById(id: Long) = try {
+      graph.getRelationshipById(id)
+    } catch {
+      case e: NotFoundException => throw new EntityNotFoundException(s"Relationship with id $id", e)
+      case e: RuntimeException  => throw e
```

Comment:
Not sure actually, I have copied the code from the node `getById`.
I can clean it up in both places and see what happens.